File: FactoryMode(抽象工厂)/FactoryMode(抽象工厂)/ViewController.m
//
//  ViewController.m
//  FactoryMode(抽象工厂)
//
//  Created by admin on 2018/8/30.
//  Copyright © 2018年 admin. All rights reserved.
//

#import "ViewController.h"
#import "WXPaymentFactory.h"
#import "ZFBPaymentFactory.h"
#import "PaymentProtocol.h"
#import "PaymentFactoryProtocol.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    id<PaymentFactoryProtocol> wxFactory = [[WXPaymentFactory alloc] init];
    id<PaymentProtocol> payment = wxFactory.getSomePayment;
    id object = payment.getPaymentObject;
    
    
   id<PaymentFactoryProtocol> zfbFactory = [[ZFBPaymentFactory alloc] init];
   id<PaymentProtocol> zfbPayment = zfbFactory.getSomePayment;
   id zfbObject = zfbPayment.getPaymentObject;
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Biulder(构建者模式)/Biulder(构建者模式)/ViewController.swift
//
//  ViewController.swift
//  Biulder(构建者模式)
//
//  Created by admin on 2018/9/8.
//  Copyright © 2018年 admin. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        
        
        
//        let alertController = ManagerController(param: )
        
        ManagerController.Biulder(self)
        .setTitle("苏维埃万岁")
        .setMessage("乌拉")
        .setConfirmText("确定")
        .setCancelText("取消")
        .setCancelBlock { (action) in
            print("点击取消")
        }
        
        .setConfirmBlock { (action) in
            print("点击确定")
        }.biuld().show()
    }


}

